File: go/variables_and_types.go
// VARIABLES AND DATA TYPES
// Demonstrates variable declaration and data types in Go

package main

import "fmt"

func main() {
	// Declare variables with explicit types
	var age int = 25
	var name string = "Alice"
	var height float64 = 5.8
	var isActive bool = true

	fmt.Println("Age:", age)
	fmt.Println("Name:", name)
	fmt.Println("Height:", height)
	fmt.Println("Is Active:", isActive)

	// Short variable declaration (type inference)
	city := "New York"
	population := 8000000
	fmt.Println("City:", city)
	fmt.Println("Population:", population)

	// Type conversion
	var x int = 10
	var y float64 = float64(x)
	fmt.Println("Converted:", y)

	// Zero values
	var emptyInt int
	var emptyString string
	var emptyBool bool
	fmt.Printf("Zero values: int=%v, string='%v', bool=%v\n", emptyInt, emptyString, emptyBool)
}

File: rust/variables_and_types.rs
// VARIABLES AND DATA TYPES
// Demonstrates variable declaration and data types in Rust

fn main() {
    // Immutable variable (default)
    let age = 25;
    let name: String = String::from("Alice");
    
    // Mutable variable
    let mut height: f64 = 5.8;
    height = 5.9;
    
    println!("Age: {}", age);
    println!("Name: {}", name);
    println!("Height: {}", height);
    
    // Primitive types
    let integer: i32 = 42;
    let floating: f32 = 3.14;
    let boolean: bool = true;
    let character: char = 'A';
    
    println!("Integer: {}", integer);
    println!("Float: {}", floating);
    println!("Boolean: {}", boolean);
    println!("Char: {}", character);
    
    // Type casting
    let x: i32 = 10;
    let y: f64 = x as f64;
    println!("Casted: {}", y);
    
    // Constants
    const PI: f64 = 3.14159;
    println!("PI: {}", PI);
    
    // Shadowing (declaring new variable with same name)
    let x = 5;
    let x = x + 1;
    println!("Shadowed x: {}", x);
}
